Recipes/Recipes.API/Controllers/RecipesController.cs

```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Recipes.Application.Services;
+
+namespace Recipes.API.Controllers;
+
+[ApiController]
+[Route("recipes")]
+public class RecipesController : ControllerBase
+{
+    private IRecipesService _recipesService;
```

Comment:
Probably this should be some RecipesRepository instead. Haven't looked through the whole source too thoroughly, but the rough idea would probably be using a [repository pattern](https://medium.com/@codebob75/repository-pattern-c-ultimate-guide-entity-framework-core-clean-architecture-dtos-dependency-6a8d8b444dcb)

Reply:
So, as I understand, we don't really need `services` in the back-end as you are already handling them? We should only have `repositories` instead?

Reply:
Yeah, I'd advocate for using just the repository as an intermediary layer between the database and the controller. Noticed these docs which seem to be more official and trustworthy, which had this nice illustration (our pathway would be the second):

![](https://asp.net/media/2578149/Windows-Live-Writer_8c4963ba1fa3_CE3B_Repository_pattern_diagram_1df790d3-bdf2-4c11-9098-946ddd9cd884.png)

Reply:
All right, I will implement this instead 👌